src/rust/engine/process_execution/src/local.rs

```diff
@@ -21,6 +23,53 @@ impl CommandRunner {
   pub fn new(store: fs::Store, fs_pool: Arc<fs::ResettablePool>) -> CommandRunner {
     CommandRunner { store, fs_pool }
   }
+
+  fn construct_output_snapshot(
+    store: Store,
+    posix_fs: Arc<fs::PosixFS>,
+    output_file_paths: BTreeSet<PathBuf>,
+    output_dir_paths: BTreeSet<PathBuf>,
+  ) -> BoxFuture<Snapshot, String> {
+    let output_dirs_glob_strings: Result<Vec<String>, String> = output_dir_paths
+      .into_iter()
+      .map(|p| {
+        p.into_os_string()
+          .into_string()
+          .map_err(|e| format!("Error stringifying output_directories: {:?}", e))
+          .map(|s| format!("{}/**", s))
+      })
+      .collect();
+
+    let output_dirs_future = posix_fs
+      .expand(
+        PathGlobs::create(
+          &try_future!(output_dirs_glob_strings),
+          &[],
+          StrictGlobMatching::Ignore,
+        ).unwrap(),
+      )
+      .map_err(|e| format!("Error stating output dirs: {}", e));
+
+    let output_files_future = posix_fs
```

Comment:
Both output files and output dirs can be collected together by doing something like:
```
let all_globs = output_dirs_glob_strings.into_iter().chain(output_file_paths.into_iter().map(|p| p.into_string_lossy()));
.. = posix_fs.expand(PathGlobs::create(all_globs, ..));
```

...but I think this might be liable to run into #5515.

Reply:
(so let's not do it here)

Reply:
👍